var/spack/repos/builtin/packages/py-pyopengl/package.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import sys
+
+from spack.package import *
+
+
+class PyPyopengl(PythonPackage):
+    """PyOpenGL is the most common cross platform Python binding to OpenGL and related APIs."""
+
+    homepage = "http://pyopengl.sourceforge.net"
+    pypi = "pyopengl/PyOpenGL-3.1.6.tar.gz"
+
+    version("3.1.6", sha256="8ea6c8773927eda7405bffc6f5bb93be81569a7b05c8cac50cd94e969dce5e27")
+
+    variant("glu", default=True, description="Enable OpenGL Utility (GLU) binding.")
+    variant("glut", default=True, description="Enable OpenGL Utility Toolkit (GLUT) binding.")
+
+    conflicts("osmesa", when="^glx")
+    conflicts("glx", when="^osmesa")
+
+    depends_on("py-setuptools", type="build")
+    # actually installing PyOpenGL itself just requires python
+    # but tests (and possibly dependent packages) need OpenGL libraries
+    depends_on("gl", type=("test", "run"))
+    depends_on("glu", when="+glu", type=("test", "run"))
+    depends_on("freeglut+shared", when="+glut", type=("test", "run"))
```

Comment:
Based on `setup_build_environment`, it looks like these should be `type="link"`, not test/run. Does the package every search `PATH` at build/run/test time, or does it only ever look in `LD_LIBRARY_PATH` to link to the library?

Reply:
`setup_build_environment` is actually not needed, it doesn't check the gl-libraries in the python setup/install script.

So I set the dependencies to "link" only and removed the `setup_build_environment`.

Does it make sense this way?

Reply:
Yep, makes more sense now.